src/audio/codec_adapter/codec_adapter.c

```diff
@@ -379,6 +379,22 @@ static int codec_adapter_copy(struct comp_dev *dev)
 	comp_dbg(dev, "codec_adapter_copy() start: codec_buff_size: %d, local_buff free: %d source avail %d",
 		 codec_buff_size, local_buff->stream.free, source->stream.avail);
 
+	if (!codec->cpd.init_done) {
+		if (bytes_to_process < codec_buff_size)
+			goto db_verify;
+
+		buffer_invalidate(source, codec_buff_size);
+		codec_adapter_copy_from_source_to_lib(&source->stream, &codec->cpd,
+						      codec_buff_size);
+		codec->cpd.avail = codec_buff_size;
+		ret = codec_init_process(dev);
+		if (ret)
+			return ret;
+
```

Comment:
Should it be here: codec->cpd.init_done = ret?
Otherwise init_done will always be null, and each time we do a codec_adapter_copy() will do a code_init_process().
Or this is the intention? Because we need to get the header information for each bytes_to_process.

Reply:
Good point, but the init_done is set by codec_init_process() function via this call:
``API_CALL(cd, XA_API_CMD_INIT, XA_CMD_TYPE_INIT_DONE_QUERY, &codec->cpd.init_done, ret)``